packages/grid/_modules_/grid/components/tools/FilterInputValue.tsx

```diff
@@ -9,11 +9,8 @@ export interface TypeFilterInputValueProps extends FilterInputValueProps {
   type?: 'text' | 'number' | 'date' | 'datetime-local';
 }
 
-export const FilterInputValue: React.FC<TypeFilterInputValueProps> = ({
-  item,
-  applyValue,
-  type,
-}) => {
+export function FilterInputValue(props: TypeFilterInputValueProps) {
```

Comment:
I think for a better typescript support, it would better to keep the type definition otherwise it might just see it as a function instead of a react functional component 🤔

Reply:
```
    interface FunctionComponent<P = {}> {
        (props: PropsWithChildren<P>, context?: any): ReactElement<any, any> | null;
        propTypes?: WeakValidationMap<P>;
        contextTypes?: ValidationMap<any>;
        defaultProps?: Partial<P>;
        displayName?: string;
    }
```

Reply:
I believe none of the added types of FunctionComponent are relevant:

- injects a `children` prop, a prop that might not be supported or even be correct: https://github.com/mui-org/material-ui/issues/23634.
- displayName is harmful most of the time
- defaultProps is on its way to deprecation
- contextTypes is legacy
- propTypes isn't relevant with TypeScript